smartsim/_core/_cli/test.py

```diff
@@ -0,0 +1,215 @@
+
+
+if t.TYPE_CHECKING:
+    # Pylint disable needed for old version of pylint w/ TF 2.6.2
+    # pylint: disable-next=unsubscriptable-object
+    _TemporaryDirectory = tempfile.TemporaryDirectory[str]
+else:
+    _TemporaryDirectory = tempfile.TemporaryDirectory
+
+
+class _VerificationTempDir(_TemporaryDirectory):
+    """A Temporary directory to be used as a context manager that will only
+    clean itself up if no error is raised within its context
+    """
+
+    def __exit__(
+        self,
+        exc: t.Optional[t.Type[BaseException]],
+        value: t.Optional[BaseException],
+        tb: t.Optional[TracebackType],
+    ) -> None:
+        if not value:  # Yay, no error! Clean up as normal
+            super().__exit__(exc, value, tb)
+        else:  # Uh-oh! Better make sure this is not implicitly cleaned up
+            self._finalizer.detach()  # type: ignore[attr-defined]
+
+
+def execute(_args: argparse.Namespace, /) -> int:
+    backends = installed_redisai_backends()
+    try:
+        with _VerificationTempDir() as temp_dir:
+            test_install(
+                location=temp_dir,
+                port=None,  # TODO: allow users to pass as arg `--port`?
+                with_tf="tensorflow" in backends,
+                with_pt="torch" in backends,
+                with_onnx="onnxruntime" in backends,
+            )
+    except Exception as e:
+        logger.error(
+            "SmartSim failed to run a simple experiment!\n"
+            f"Experiment failed due to the following exception:\n{e}\n\n"
+            f"Output files are available at `{temp_dir}`"
+        )
+        return 2
+    return 0
+
+
+def test_install(
+    location: str,
+    port: t.Optional[int],
+    with_tf: bool,
+    with_pt: bool,
+    with_onnx: bool,
+) -> None:
+    exp = Experiment("TestExperiment", exp_path=location, launcher="local")
+    port = port or _find_free_port()
+    with _make_managed_local_orc(exp, port) as client:
+        logger.info("Verifying Tensor Transfer")
+        client.put_tensor("plain-tensor", np.ones((1, 1, 3, 3)))
+        client.get_tensor("plain-tensor")
+        if with_tf:
+            logger.info("Verifying TensorFlow Backend")
+            _test_tf_install(client)
+        if with_pt:
+            logger.info("Verifying Torch Backend")
+            _test_torch_install(client)
+        if with_onnx:
+            logger.info("Verifying ONNX Backend")
+            _test_onnx_install(client)
+
+
+@contextmanager
+def _make_managed_local_orc(
+    exp: Experiment, port: int
+) -> t.Generator[Client, None, None]:
+    """Context managed orc that will be stopped if an exception is raised"""
+    orc = exp.create_database(db_nodes=1, interface="lo", port=port)
+    exp.generate(orc)
+    exp.start(orc)
+    try:
+        (client_addr,) = orc.get_address()
+        yield Client(address=client_addr, cluster=False)
+    finally:
+        exp.stop(orc)
+
+
+def _find_free_port() -> int:
```

Comment:
Interesting approach. Maybe in the future we could think of implementing this for the test ports (in the test suite) too?

Reply:
I'd be more than willing to add something like this to the test suite! It's defiantly not the most robust solution, but certainly should be "good enough" for our test suite!

I'll write up a ticket!!